jabgui/src/main/java/org/jabref/gui/preview/ClipboardContentGenerator.java

```diff
@@ -123,25 +123,60 @@ static ClipboardContent processHtml(List<String> citations) {
     }
 
     /**
-     * Insert each citation into HTML.
-     * convert HTML to markdown using flexmark.
-    */
+     * Converts a list of citations to Markdown.
+     *
+     * Behavior:
+     * - If the citations appear to be HTML (e.g., CSL HTML output), convert via Flexmark (HTML -> Markdown).
+     * - Otherwise, assume the citations are already Markdown (e.g., CSL Markdown output) and just join them.
+     */
     @VisibleForTesting
     static ClipboardContent processMarkdown(List<String> citations) {
-        String result = "<!DOCTYPE html>" + OS.NEWLINE +
-                "<html>" + OS.NEWLINE +
-                "   <head>" + OS.NEWLINE +
-                "      <meta charset=\"utf-8\">" + OS.NEWLINE +
-                "   </head>" + OS.NEWLINE +
-                "   <body>" + OS.NEWLINE + OS.NEWLINE;
+        // Default behavior (non-CSL-aware): detect HTML vs Markdown by content
+        return processMarkdown(citations, false);
+    }
 
-        result += String.join(CitationStyleOutputFormat.HTML.getLineSeparator(), citations);
-        result += OS.NEWLINE +
-                "   </body>" + OS.NEWLINE +
-                "</html>" + OS.NEWLINE;
+    /**
+     * Converts a list of citations to Markdown.
+     *
+     * If fromCitationStyle is true, we assume the list items are already Markdown (produced by citeproc-java with
+     * output format MARKDOWN) and simply join them with new lines. Otherwise, we fall back to HTML detection and
+     * Flexmark-based conversion when needed.
+     */
+    @VisibleForTesting
+    static ClipboardContent processMarkdown(List<String> citations, boolean fromCitationStyle) {
```

Comment:
Please no boolean parameters for method variants

Introduce two differently named methods (maybe with suffix): `processMarkdownCslMode`, `processMarkdownAutodectMode`